### ISSUE TITLE
License clarifications

### ISSUE THREAD

**Issue by stefanistrate** [DBD-research-group/BirdSet#337]
Hello!

I have found some inconsistencies between what the README at https://huggingface.co/datasets/DBD-research-group/BirdSet says about data licensing, and what the individual recordings say.

For the XCL `train` split I'm seeing this distribution:

```
License                                           Clips
-------------------------------------------------------
nan                                              122531
//creativecommons.org/licenses/by-nc-sa/4.0/       8439
//creativecommons.org/publicdomain/zero/1.0/         74
//creativecommons.org/licenses/by-sa/4.0/            47
//creativecommons.org/licenses/by-nc-sa/3.0/         13
//creativecommons.org/licenses/by/4.0/               10
//creativecommons.org/licenses/by-nc/4.0/             7
TOTAL                                            131121
```

For all cumulated `test_5s` splits I'm seeing this distribution:

```
License                                                            Clips
------------------------------------------------------------------------
Creative Commons Attribution 4.0 International Public License     101685
Creative Commons Zero v1.0 Universal                                4414
None                                                                  62
TOTAL                                                             106161
```

1. Can you clarify where the `nan` and missing licenses (= `None`) are coming from?
2. The README says:
    > Each recording in the training datasets has a unique recordist and the corresponding license from XC. We omit all recordings from XC that are CC-ND.

    The lack of a license (i.e. `nan` or `None`) is a stronger restriction than CC-ND. Can you confirm that all training recordings can be used in a non-commercial way?
3. The README says:
    > We exclude all recordings with a SA licenses. Every recording is NC.

    I'm still seeing recordings with CC-BY-SA and CC-BY-NC-SA. Is this an inconsistency, and those recordings were supposed to be filtered out in the first place?
4. The README says:
    > Each test dataset is licensed under CC BY 4.0.

    Please clarify the missing licenses (= `None`) from the `test_5s` split.
 
Thanks,
Ștefan

**Comment by lurauch** [comment]
Hi Ștefan,

Good catch on the `XCL` metadata. We've confirmed a conversion issue: there should be no `none` licenses, and `by-nc-sa/4.0` should account for 400k+ samples. We’re fixing this now. Could you please verify your local XCL count? The full set should contain 528,422 files according to the metadata.

Regarding the test dataset, could you specify which test set showed the `none` license? I guess `NBP` because there also seems to be a conversion issue. Note that there should also be about 539 test_5s files in `NBP`. 

Thanks!

**Comment by stefanistrate** [comment]
Thanks for addressing this issue. I forgot to say that the counts I've posted (`XCL` and `test_5s`) were obtained after I've filtered the data to a list of target species, so you are right, the full counts are larger than my numbers.

Let me know when you push a new version to HF so that I can verify the licenses again.

Ștefan

**Comment by lurauch** [comment]
@Moritz-Wirth is taking care of it. He'll let you know, it should be done very soon it's just an update of the metadata file :)

**Comment by Moritz-Wirth** [comment]
Hi Ștefan,

Thanks for catching that. I’ve added the missing license metadata to all training and test files. If you’re using Hugging Face datasets, the metadata should update automatically.

To clarify what the README intended to say: we only excluded the NB files. Initially we had discussed also excluding all SA files (which would have removed the majority of the data), and that earlier plan mistakenly ended up in the text. I’ve now corrected the README to reflect the actual exclusion.

Here are the correct license counts for the training set (XCM; XCL below):
```
license
//creativecommons.org/licenses/by-nc-sa/4.0/    76268
//creativecommons.org/licenses/by-nc-sa/3.0/    11858
//creativecommons.org/licenses/by-sa/4.0/        1085
//creativecommons.org/licenses/by-sa/3.0/         330
//creativecommons.org/publicdomain/zero/1.0/      193
//creativecommons.org/licenses/by-nc/4.0/          32
//creativecommons.org/licenses/by/4.0/             29
```

Let me know if anything else looks off.

**Comment by lurauch** [comment]
I think @Moritz-Wirth gave you the license counts for `XCM `not `XCL`. These are the correct counts: 

| license | count |
|---|---:|
| `//creativecommons.org/licenses/by-nc-sa/4.0/` | 450630 |
| `//creativecommons.org/licenses/by-nc-sa/3.0/` | 69076 |
| `//creativecommons.org/licenses/by-sa/4.0/` | 6725 |
| `//creativecommons.org/publicdomain/zero/1.0/` | 920 |
| `//creativecommons.org/licenses/by-sa/3.0/` | 680 |
| `//creativecommons.org/licenses/by/4.0/` | 220 |
| `//creativecommons.org/licenses/by-nc/4.0/` | 170 |
| `//creativecommons.org/licenses/by-nc-sa/3.0/us/` | 1 |

Additionally, he meant that we exclude `ND`, not `NB `files :)

**Comment by stefanistrate** [comment]
I can confirm the training numbers. Things look good on my side as well. I am using Hugging Face indeed, but I had to manually delete its cache from my computer to force the new metadata download.

One other suggestion I would make (but it's not blocking me at this stage): it would be nice to consolidate the license strings in `test`, `test_5s` and `valid`, to be consistent with the `train` split. At the moment I'm seeing strings like:

- `Creative Commons Attribution 4.0 International Public License`
- `Creative Commons Zero v1.0 Universal`
- `//creativecommons.org/licenses/by/4.0/`

while the `train` split only has `//creativecommons.org/...`.

**Comment by lurauch** [comment]
Good suggestion. We'll include this in the next BirdSet iteration!
